tools/codegen/api/lazy.py

```diff
@@ -108,14 +132,15 @@ def __init__(self, func: FunctionSchema):
                           "tensor_options",
                           "post_tensor_options_kwarg_only",
                           "out"]:
-            if getattr(func.arguments, arg_field) is not None:
-                keyword_arg_types.extend([
-                    NamedCType(
-                        arg.name,
-                        process_ir_type(arg.type)) for arg in getattr(func.arguments, arg_field)])
+            curr_args = getattr(func.arguments, arg_field)
+            if curr_args is not None:
+                if isinstance(curr_args, TensorOptionsArguments):
+                    curr_args = curr_args.all()
+                keyword_arg_types.extend([NamedCType(arg.name, process_ir_type(arg.type)) for arg in curr_args])
         self.keyword_arg_types = tuple(keyword_arg_types)
         self.name = func.name
         self.returns = func.returns
+        self.wrapped_scalar_names = [arg.name for arg in func.schema_order_arguments() if isWrappedScalarType(arg.type)]
```

Comment:
Ok, so the `LazyIrSchema` object knows "what are the C++ types and names of all arguments in my schema", and now it also knows "which of my arguments were originally scalars before processing", using this separate list.

I think the other "data-model-y" option you could take would be to represent that info per-argument using a new type (e.g. a `LTCArgument` that knows both its `NamedCType`, and whether it was a scalar argument). The advantage there is that your processing functions (`node_ctor_arg_rvalue_string`) can run on each argument in isolation, instead of having to take in the whole schema. But if this list is used in a really minimal/targeted way then maybe that's overkill.

Reply:
i think this approach sounds good.  generally i think i want to separate out more 'refactoring' to a separate diff.  Lets collect ideas for that on a GH issue. i'll make one and link here: https://github.com/pytorch/pytorch/issues/72960